### DISCUSSION TITLE
How to make new `variant` available in typescript suggestion

### DISCUSSION THREAD

**Discussion by minhchu** [chakra-ui/chakra-ui#5720] [Q&A] [answered]
Hi, I've just made a new Table variant like this:

```ts
const borderedVariant: PartsStyleFunction<typeof parts> = (props) => {
  const { colorScheme: c } = props;

  return {
    table: {
      borderSpacing: 0,
      border: '1px',
      borderColor: `${c}.400`,
    },
  };
};

export const TableComponent = {
  components: {
    Table: {
      variants: {
        bordered: borderedVariant,
      },
    },
  },
};

```
then in my component:

```ts
const List = () => (
   <Table variant="bordered"></Table>
)
```

And here is the suggestion list:
![Screenshot from 2022-03-11 19-56-23](https://user-images.githubusercontent.com/14136809/157871071-b5d86d05-d475-4339-ba51-a23b75a21277.png)
As you can see there is no `bordered`.

The question is how can I make the new `bordered` variant available in the suggestion ?

**Comment by chr-ge** [comment, marked as answer]
Use `@chakra/cli` to generate your custom [theme-typings](https://chakra-ui.com/docs/styled-system/theming/advanced#theme-typings).
